addons/binding/org.openhab.binding.sony/src/main/java/org/openhab/binding/sony/internal/net/SocketChannelSession.java

```diff
@@ -0,0 +1,420 @@
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.nio.ByteBuffer;
+import java.nio.channels.AsynchronousCloseException;
+import java.nio.channels.SocketChannel;
+import java.util.List;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Represents a restartable socket connection to the underlying telnet session. Commands can be sent via
+ * {@link #sendCommand(String)} and responses will be received on any {@link SocketSessionListener}. This implementation
+ * of {@link SocketSession} communicates using a {@link SocketChannel} connection.
+ *
+ * @author Tim Roberts
+ */
+public class SocketChannelSession implements SocketSession {
+    private Logger _logger = LoggerFactory.getLogger(SocketChannelSession.class);
+
+    /**
+     * The host/ip address to connect to
+     */
+    private final String _host;
+
+    /**
+     * The port to connect to
+     */
+    private final int _port;
+
+    /**
+     * The actual socket being used. Will be null if not connected
+     */
+    private final AtomicReference<SocketChannel> _socketChannel = new AtomicReference<SocketChannel>();
+
+    /**
+     * The {@link ResponseReader} that will be used to read from {@link #_readBuffer}
+     */
+    private final ResponseReader _responseReader = new ResponseReader();
+
+    /**
+     * The responses read from the {@link #_responseReader}
+     */
+    private final BlockingQueue<Object> _responses = new ArrayBlockingQueue<Object>(50);
+
+    /**
+     * The dispatcher of responses from {@link #_responses}
+     */
+    private final Dispatcher _dispatcher = new Dispatcher();
+
+    /**
+     * The {@link SocketSessionListener} that the {@link #_dispatcher} will call
+     */
+    private List<SocketSessionListener> _listeners = new CopyOnWriteArrayList<SocketSessionListener>();
+
+    /**
+     * Creates the socket session from the given host and port
+     *
+     * @param host a non-null, non-empty host/ip address
+     * @param port the port number between 1 and 65535
+     */
+    public SocketChannelSession(String host, int port) {
+        if (host == null || host.trim().length() == 0) {
+            throw new IllegalArgumentException("Host cannot be null or empty");
+        }
+
+        if (port < 1 || port > 65535) {
+            throw new IllegalArgumentException("Port must be between 1 and 65535");
+        }
+        _host = host;
+        _port = port;
+    }
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see
+     * org.openhab.binding.russound.internal.net.SocketSession#addListener(org.openhab.binding.russound.internal.net.
```

Comment:
remove the russound references - probably best to completely remove the JavaDoc on overridden methods as they do not bring much value.